[Glimmer2] Closure component test migration

Serabe opened the pull request:
Migrate tests for closure component.

Serabe committed:
[Glimmer2] Closure component test migration

Migrate tests for closure component.

Serabe commented:
@chancancode I think these are finished. There are a couple of tests skipped and some parts of other commented because they are related to #13158

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText('Hodi Sigmundur 33');
+  }
+
+  ['@skip bound outer named parameters get updated in the right scope']() {
```
Does this work with `@htmlbars`?

Serabe replied:
No, because of the issue with updating positional parameters with the `component` helper.

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText('Inner 28');
+  }
+
+  ['@skip bound outer hash parameters get updated in the right scope']() {
```
Same question.

Serabe replied:
Same answer. In other cases, we turn to commenting out the UR part, should I proceed with that?

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+
+    this.assertText('Hodi Sergio');
+
+    this.runTask(() => this.context.set('name', 'Hodari'));
```
Miss `this.assertText('Hodi Hodari');`

Serabe replied:
Fixed for the next push.

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+  ['@test renders with dot path and attr']() {
+    let expectedText = 'Hodi';
+    this.registerComponent('-looked-up', {
+      template: expectedText
```
This should be `{{expectedText}}`.

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText(expectedText);
+
+    /*
+    this.runTask(() => this.context.set('expectedText', 'Hola'));
```
Remove?

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+  ['@htmlbars renders with dot path and curried over attr']() {
+    let expectedText = 'Hodi';
+    this.registerComponent('-looked-up', {
+      template: expectedText
```
Should be`{{expectedText}}`

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText('Hola');
+    */
+
+    this.runTask(() => this.context.set('expectedText', expectedText));
```
Missing `this.assertText(expectedText);`

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText(expectedText);
+
+    /*
+    this.runTask(() => this.context.set('expectedText', 'Hola'));
```
We should be do U.R. steps here

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+    this.assertText('Hola');
+    */
+
+    this.runTask(() => this.context.set('expectedText', expectedText));
```
Missing this.assertText(expectedText);

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,669 @@
+      myProp: 1
+    });
+
+    assert.equal(this.$('#nested-prop').text(), '1');
```
Will defer to @chancancode, but not sure we are trying to move away from jQuery in tests.

Serabe replied:
That `this.$` is defined in [`abstract-test-case`](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/utils/abstract-test-case.js#L141-L143). If we were to move away from jQuery, we would just need to change things there.

chadhietala replied:
Sounds good.

Serabe commented:
@chadhietala thank you so much for reviewing this! I addressed most of the issues, leaving only the jQuery one (commented [here](https://github.com/emberjs/ember.js/pull/13214/files#r59298810)) and the skipped tests (commented [here](https://github.com/emberjs/ember.js/pull/13214/files#r59297948)).

Thank you one more time!

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  }
-});
-
-QUnit.test('renders with component helper', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R10)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), expectedText, '-looked-up component rendered');
-});
-
-QUnit.test('renders with component helper with invocation params, hash', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R26)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Hodi Hodari', '-looked-up component rendered');
-});
-
-QUnit.test('renders with component helper with curried params, hash', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R45)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Hola Hodari', '-looked-up component rendered');
-});
-
-QUnit.test('updates when component path is bound', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R65)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-        'hindi lookupComponent renders greeting');
-});
-
-QUnit.test('updates when curried hash argument is bound', function() {
```
This test doesn't look like it was migrated.

Serabe replied:
Will be added in the next pushed. It might take a couple of hours to modify every test.

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,671 @@
+    });
+
+    this.render('{{component (component lookupComponent)}}', {
+      lookupComponent: '-mandarin'
```
Can you change this to

```
this.render('{{component (component lookupComponent)}}', {
  model: { lookupComponent: '-mandarin' }
});
```

This way we can actually do a replacement in the R step.

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,671 @@
+      template: 'Namaste'
+    });
+
+    this.render('{{component (component lookupComponent)}}', {
```
This would become `{{component (component model.lookupComponent)}}`

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,671 @@
+
+    this.assertText('Namaste');
+
+    this.runTask(() => this.context.set('lookupComponent', '-mandarin'));
```
This would be become `this.runTask(() => this.context.set('model', { lookupComponent: '-mandarin' }));`

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,671 @@
+
+    this.assertText('ni hao');
+
+    this.runTask(() => this.context.set('lookupComponent', '-hindi'));
```
This would become `this.runTask(() => this.context.set('model.lookupComponent', '-hindi'));`

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-        'greeting is bound');
-});
-
-QUnit.test('updates when curried hash arguments is bound in block form', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R93)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-        'greeting is bound');
-});
-
-QUnit.test('nested components overwrites named positional parameters', function() {
```
Seems to be lost in the migration.

Serabe replied:
This [one](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R118)

chadhietala replied:
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R151)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Hodari 21', '-looked-up component rendered');
-});
-
-QUnit.test('nested components overwrites hash parameters', function() {
```
Seems to be lost in the migration.

chadhietala replied:
NM

✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R168)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Inner 28', '-looked-up component rendered');
-});
-
-QUnit.test('conflicting positional and hash parameters raise and assertion if in the same closure', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R245)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  }, `You cannot specify both a positional param (at position 0) and the hash argument \`name\`.`);
-});
-
-QUnit.test('conflicting positional and hash parameters does not raise and assertion if rerendered', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R258)

chadhietala commented on packages/ember-glimmer/tests/integration/components/closure-components-test.js:
```diff
@@ -0,0 +1,671 @@
+    this.assertText('Hodi Hodari');
+  }
+
+  ['@test conflicting positional and has parameters does not raise an assertion if in different closure']() {
```
`has` -> `hash`

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Hodi Sergio', 'component is rendered');
-});
-
-QUnit.test('conflicting positional and hash parameters does not raise and assertion if in the different closure', function() {
```
✔️  [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R287)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Hodi Sergio', 'component is rendered');
-});
-
-QUnit.test('raises an assertion when component path is null', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R304)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  });
-});
-
-QUnit.test('raises an assertion when component path is not a component name', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R310)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  }, `The component helper cannot be used without a valid component name. You used "not-a-component" via (component compName)`);
-});
-
-QUnit.test('renders with dot path', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R324)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), expectedText, '-looked-up component rendered');
-});
-
-QUnit.test('renders with dot path and attr', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R342)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), expectedText, '-looked-up component rendered');
-});
-
-QUnit.test('renders with dot path curried over attr', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R371)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), expectedText, '-looked-up component rendered');
-});
-
-QUnit.test('renders with dot path and with rest positional parameters', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R400)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), `${expectedText},Hola`, '-looked-up component rendered with rest params');
-});
-
-QUnit.test('renders with dot path and rest parameter does not leak', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R433)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  ok(isEmpty(value), 'value is an empty parameter');
-});
-
-QUnit.test('renders with dot path and updates attributes', function() {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R458)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$('#nested-prop').text(), '3', 'value got updated again');
-});
-
-QUnit.test('adding parameters to a closure component\'s instance does not add it to other instances', function(assert) {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R509)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  equal(component.$().text(), 'Foo', 'there is only one Foo');
-});
-
-QUnit.test('parameters in a closure are mutable when closure is a param', function(assert) {
```
✔️ [Migrated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R533)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  runDestroy(dispatcher);
-});
-
-QUnit.test('parameters in a closure are mutable when closure is in a nested param', function(assert) {
```
[Consolidated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R624)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  runDestroy(dispatcher);
-});
-
-QUnit.test('parameters in a closure are mutable when closure is a hash value', function(assert) {
```
[Consolidated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R624)

chadhietala commented on packages/ember-htmlbars/tests/helpers/closure_component_test.js:
```diff
@@ -1,776 +0,0 @@
-  runDestroy(dispatcher);
-});
-
-QUnit.test('parameters in a closure are mutable when closure is a nested hash value', function(assert) {
```
[Consolidated](https://github.com/emberjs/ember.js/pull/13214/files#diff-8e2359cbe9c551044787f512fe265932R624)

chadhietala commented:
@Serabe Looks like there was only one missing test. Can you please update all of the R steps so that we actually do the replacement. See my comment here https://github.com/emberjs/ember.js/pull/13214#discussion-diff-60105331 and #13325 for details.

Serabe commented:
@chadhietala thank you so much again for taking time to review this. I've just uploaded the changes. Let me know if I forgot something or if I can do anything to improve this, let me know! Thank you!

Serabe commented:
Rebased master. I had not done so in a while :$

chadhietala commented:
@Serabe Awesome! I think this one is good to go. 🎉 💥 🎉 .

rwjblue commented:
Thank you @Serabe!
